### PR TITLE
update stac-fastapi to next major version

### PR DESCRIPTION
**Pull request by vincentsarago**
For visibility purpose

This PR aims to update stac-fastapi-pgstac to the next stac-fastapi major version (3.0)

### CODE REVIEW

**Comment by vincentsarago** [comment]
Tests are failing right now because there was a change in the transaction extension 

````
# stac-fastapi 2.5
            endpoint=create_async_endpoint(
                self.client.create_collection, stac_types.Collection
            ),

# stac-fastapi 3.0
           endpoint=create_async_endpoint(self.client.create_collection, Collection),
```

This means that the input body is now validated and will be a Pydantic Object, with 2.5 the model was a Typed dict

**Comment by vincentsarago** [comment]
😬 I don't understand, tests passes here but not locally 🤦

**Comment by jonhealy1** [comment]
So weird. The queryables and PUT /collections tests are both failing for me ...

File: stac_fastapi/pgstac/extensions/filter.py [hunk 1]
```diff
@@ -37,5 +39,4 @@ async def get_queryables(
                 raise NotFoundError(f"Collection {collection_id} not found")
 
             queryables["$id"] = str(request.url)
-            headers = {"Content-Type": "application/schema+json"}
-            return JSONResponse(queryables, headers=headers)
+            return queryables
```
Comment (vincentsarago) [inline stac_fastapi/pgstac/extensions/filter.py:42]: the response class is defined here https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/extensions/stac_fastapi/extensions/core/filter/filter.py#L108

**Comment by jonhealy1** [comment]
pre-release sounds good

**Comment by vincentsarago** [comment]
## ToDo

- remove ContextExtension
- update to pgstac 0.9.x

**Comment by jonhealy1** [comment]
Changelog maybe?

**Comment by jonhealy1** [comment]
I copied this strategy for sfeos as well - finally got the tests to pass - https://github.com/stac-utils/stac-fastapi-elasticsearch-opensearch/pull/234

**Comment by vincentsarago** [comment]
> Are we going to release an alpha version of stac-fastapi-pgstac or are we going to release v3.0.0 of stac-fastapi first?

I don't think there is any rush publishing an alpha version here! I think I'm going to focus on finishing reviewing the PR on the different repos and also finish the PR in stac-pydantic first.

**Comment by jonhealy1** [comment]
True makes sense

**Pull request merged**